### PR TITLE
Feature/ai 63/implement tools

### PR DESCRIPTION
**Pull request by simdage**

### CODE REVIEW

File: croptalk/retriever.py [hunk 1]
```diff
@@ -75,7 +75,7 @@ class RetrieverInput(BaseModel):
     county: Optional[str] = Field(description="County name. Example: Ventura")
 
 
-def format_docs(docs: Sequence[Document]) -> str:
+def format_docs(docs: Sequence[Document]) -> List[str]:
```
Comment (dtourillon-cropguard) [inline croptalk/retriever.py:78]: Is this function still used??
It seems to be using the chromadb retriever, no?

argh... we need a bit of a cleanup I think!

File: install_dsmain.sh [hunk 2]
```diff
@@ -25,7 +25,4 @@ git clone git@github.com:cropguard-ai/ds-main.git
 
 # Install the repo
 cd ds-main
-pip install . -v
-
-# TODO: add ds-main as a dependency in chat-langchain's pyproject.toml
```
Comment (dtourillon-cropguard) [inline install_dsmain.sh:30]: Why did we remove the TODO exactly? I may be wrong, but I think the comment is still relevant

File: _scripts/evaluate_tools.py [hunk 3]
```diff
@@ -0,0 +1,71 @@
...
+import json
+from langchain_core.tracers.context import tracing_v2_enabled
+from croptalk.tools import tools
+from langchain.tools.render import render_text_description
+
+logging.basicConfig(level=logging.INFO)
+logger = logging.getLogger()
+
+
+def parse_args() -> argparse.Namespace:
```
Comment (dtourillon-cropguard) [inline _scripts/evaluate_tools.py]: Is this fct called anywhere? If not, I'd remove it

File: _scripts/evaluate_tools.py [hunk 4]
```diff
@@ -0,0 +1,71 @@
...
+    )
+    parser.add_argument(
+        "eval_path",
+        help="CSV file path that contains evaluation use cases",
+    )
+    return parser.parse_args()
+
+
+if __name__ == "__main__":
+    # parse args
```
Comment (dtourillon-cropguard) [inline _scripts/evaluate_tools.py]: remove comment?

File: croptalk/model_llm.py [hunk 5]
```diff
@@ -66,19 +74,74 @@ def create_retriever_chain(llm: BaseLanguageModel, document_retriever: DocumentR
...
+            chosen_tool = tool_map[model_output["name"]]
+            return itemgetter("arguments") | chosen_tool
+        except Exception as e:
+            return "NO_TOOL_ANSWER"
+
+    tool_chain = prompt_tool | llm | JsonOutputParser() | tool_pipe | StrOutputParser()
+
+    prompt2 = PromptTemplate.from_template(
+        """
+        You are a helpful assistant. Answer the provided question : {question}. Knowing that the this answer was
```
Comment (dtourillon-cropguard) [inline croptalk/model_llm.py]: "
```suggestion
        You are a helpful assistant. Answer the provided question : {question}. Knowing that this answer was
```

File: croptalk/tools.py [hunk 6]
```diff
@@ -1,21 +1,88 @@
...
+    )
+
+
+@tool("wfrp-commodities-tool")
+def get_wfrp_commodities(reinsurance_yr: str, state_code: str, county_code: str) -> Optional[str]:
+    """
+    This tool is used to find commodities associated with the Whole Farm Revenue Protection (WFRP) insurance program
+     for a given reinsurance year, state code and county code.
+
+    To use this tool, you must provide all three arguments [reinsurance_ur, state_code and county_code]
```
Comment (dtourillon-cropguard) [inline croptalk/tools.py]:
```suggestion
    To use this tool, you must provide all three arguments [reinsurance_yr, state_code and county_code]
```

File: croptalk/retriever.py (hunk 1 above)
Comment (simdage) [reply croptalk/retriever.py:78]: indeed, I was just looking at the code and made a few changes

File: install_dsmain.sh (hunk 2 above)
Comment (simdage) [reply install_dsmain.sh:30]: Good point, we need to create a task in JIRA to keep track of it. Altough this formating was providing from reading the file.

**Pull request merged**